src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -812,11 +812,8 @@ async Task<IEnumerable<IAccount>> ITokenCacheInternal.GetAccountsAsync(Authentic
             bool filterByClientId = !_featureFlags.IsFociEnabled;
             bool isAadAuthority = requestParameters.AuthorityInfo.AuthorityType == AuthorityType.Aad;
 
-            // this will either be the home account id or null, it can never be obo assertion or tenant id
-            string partitionkey = SuggestedWebCacheKeyFactory.GetKeyFromRequest(requestParameters);
```

Comment:
GetAccountsAsync serves 2 high level APIs: 

1. GetAccounts() - i.e. get ALL accounts
2. GetAccount(string homeAccountId) - i.e. get 1 account, for which I know the ID

The second version is what CCA uses and what we need to optimize. The first one is deprecated on CCA and we only use it on PCA. 

As such, i think your code was actually correct, but we definitely need tests around it.

Reply:
Added the partition key back. Tested a bit and GetAccount and GetAccounts behavior seems to be correct.